compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -100,6 +101,25 @@ protected void populateCompilerContext() {
         options.put(PROJECT_DIR, this.sourceRoot().toAbsolutePath().toString());
     }
 
+    /**
+     * Refresh the project to clear compilation caches.
+     */
+    public void refresh() {
+        cloneProject(this);
+        CompilerContext compilerContext = this.projectEnvironmentContext()
```

Comment:
Does the`CompilerContext` holds any compilation caches such as langlib etc?

Reply:
Yes it does. The caches other than the langlibs are cleared upon refresh. Works similar to edit operations.